arm-resources/resources/2017-05-10/swagger/resources.json

```diff
@@ -0,0 +1,2727 @@
+      },
+      "required": [
+        "properties"
+      ],
+      "description": "Deployment operation parameters."
+    },
+    "DeploymentExportResult": {
+      "properties": {
+        "template": {
+          "type": "object",
+          "description": "The template content."
+        }
+      },
+      "description": "The deployment export result. "
+    },
+    "ResourceManagementErrorWithDetails": {
+      "properties": {
+        "code": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The error code returned when exporting the template."
+        },
+        "message": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The error message describing the export error."
+        },
+        "target": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The target of the error."
+        },
+        "details": {
+          "readOnly": true,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ResourceManagementErrorWithDetails"
+          },
+          "description": "Validation error."
+        }
+      },
+      "description": "The detailed error message of resource management."
+    },
+    "AliasPathType": {
+      "properties": {
+        "path": {
+          "type": "string",
+          "description": "The path of an alias."
+        },
+        "apiVersions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The API versions."
+        }
+      },
+      "description": "The type of the paths for alias. "
+    },
+    "AliasType": {
+      "properties": {
+        "name": {
+          "type": "string",
+          "description": "The alias name."
+        },
+        "paths": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AliasPathType"
+          },
+          "description": "The paths for an alias."
+        }
+      },
+      "description": "The alias type. "
+    },
+    "ProviderResourceType": {
+      "properties": {
+        "resourceType": {
+          "type": "string",
+          "description": "The resource type."
+        },
+        "locations": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The collection of locations where this resource type can be created."
+        },
+        "aliases": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/AliasType"
+          },
+          "description": "The aliases that are supported by this resource type."
+        },
+        "apiVersions": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "The API version."
+        },
+        "properties": {
+          "type": "object",
+          "additionalProperties": {
+            "type": "string",
```

Comment:
You can't really use the `ResourceGroup` of PUT for PATCH since that has a required parameter.
Storage team invents an update parameter for this PATCH operation: https://github.com/Azure/azure-rest-api-specs/blob/master/arm-storage/2016-12-01/swagger/storage.json#L200. For developers who are using PATCH to update, they only need to set the fields that require modifications. Being able to reuse the object from PUT or GET doesn't buy them much advantage.